src/utils/apiResponse.utils.ts

```diff
@@ -0,0 +1,29 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+export type ControllerResponse = {
+  isSuccess: number;
+  statusCode: number;
+  message: string;
+  data?: any;
+  error?: any;
+};
+
+export const APIResponse = {
+  create: (
+    isSuccess: number,
+    statusCode: number,
+    message?: string,
+    data?: any,
+    error?: any,
+  ): ControllerResponse => {
+    const response: ControllerResponse = { isSuccess, statusCode, message };
+    if (error) {
+      response.error = error;
+    } else if (!error && data) {
+      response.data = data;
+    } else {
+      response.data = {};
+    }
```

Comment:
PullProofAI

The `APIResponse.create` function can be improved by simplifying the conditional logic for setting `data` and `error`.
```diff
    if (error) {
      response.error = error;
    } else if (data) {
      response.data = data;
    } else {
      response.data = {};
``` 

By making these changes, the code will become more readable, maintainable, and self-explanatory. It enhances the clarity of the response structure and simplifies the logic for handling data and errors in the API responses.


<!-- This is an auto-generated comment by PullProofAI -->